src/utils/Block.ts

```diff
@@ -0,0 +1,332 @@
+import { nanoid } from 'nanoid';
+import { EventBus } from './EventBus';
+import refElementsCollection from './RefElementsCollection';
+
+// Нельзя создавать экземпляр данного класса
+class Block {
```

Comment:
Вообще, на самом деле, неплохо было бы типизировать этот класс с помощью дженериков для пропсов. Чтобы из дочерних классов указывать с какими пропсами работать. Это желательно, но не обязательно. Можешь посмотреть как это реализовано в реакте и сделать похожим образом
```suggestion
class Block<TProps> {
```